src/ucan.ts

```diff
@@ -212,61 +222,33 @@ export interface Capability<
  */
 export type Constraints<C extends Capability> = Omit<C, "can" | "with">
 
-/**
- * A string-endcoded decentralized identity document (DID).
- */
-export type DID<T = unknown> = ToString<T, `did:${string}`>
-
-/**
- * A byte-encoded {@link DID} that provides a `did` accessor method (see {@link Identity}).
- */
-export interface DIDView extends ByteView<DID>, Identity {}
-
 export type CIDVersion = 0 | 1
 
 /**
  * Represents an IPLD link to a specific data of type `T`.
+ * 
+ * Note: this extends MultiformatsCID until multiformats 10 is shipped
  *
- * @template T logical type of the data being linked to. This is distinct from the multicodec code of the underlying {@link CID}, which is represented by `C`.
+ * @template Data - Logical type of the data being linked to.
+ * @template Format - multicodec code corresponding to a codec linked data is encoded with
+ * @template Alg - multicodec code corresponding to the hashing algorithm of the CID
  * @template V - CID version
- * @template C - multicodec code corresponding to a codec linked data is encoded with
- * @template A - multicodec code corresponding to the hashing algorithm of the CID
  */
-
-export interface Link<
-  T extends unknown = unknown,
-  C extends number = number,
-  A extends number = number,
+export interface IPLDLink<
+  Data extends unknown = unknown,
+  Format extends number = number,
+  Alg extends number = number,
   V extends CIDVersion = 1
-> extends CID<C, A, V>,
-    Phantom<T> {}
-
-/**
- * Logical representation of *C*ontent *Id*entifier with optional type parameters
- * to capture the CID version, hash algorithm, and content encoding (multicodec) of the
- * identified content.
- *
- * Note: This is not an actual definition from js-multiformats because that one
- * refers to a specific class and therefore is problematic.
- *
- * @see https://github.com/multiformats/js-multiformats/pull/161  which will likely
- * replace this definition once merged.
- *
- * @template C - multicodec code corresponding to a codec content was encoded in
- * @template A - multicodec code corresponding to the hashing algorithm used to derive CID
- * @template V - CID version
- */
-export interface CID<
-  C extends number = number,
-  A extends number = number,
-  V extends CIDVersion = CIDVersion
-> {
+  > extends Phantom<Data>, MultiformatsCID {
   readonly version: V
-  readonly code: C
-  readonly multihash: MultihashDigest<A>
-  readonly bytes: Uint8Array
+  readonly code: Format
+  readonly multihash: MultihashDigest<Alg>
+
+  readonly byteOffset: number
+  readonly byteLength: number
+  readonly bytes: ByteView<IPLDLink<Data, Format, Alg, V>>
 
-  // readonly asCID: this
```

Comment:
This was used in bunch of places & also a reason why I could not extend `CID` class because it's marked private there and that caused TS to complain.

I don't mind adding `//@ts-expect-error` in those places if this is really important, but just want to make sure you're aware of this.